indico/modules/attachments/controllers/event_package.py

```diff
@@ -144,11 +169,13 @@ def _generate_zip_file(self, attachments):
         # done sending it to the client.
         delete_file.apply_async(args=[temp_file.name], countdown=3600)
         temp_file.delete = False
-        return send_file('material-{}.zip'.format(self._conf.id), temp_file.name, 'application/zip', inline=False)
+        return send_file('material-{}.zip'.format(self.event_new.id), temp_file.name, 'application/zip', inline=False)
 
     def _prepare_folder_structure(self, attachment):
-        event_dir = secure_filename(self._conf.getTitle(), None)
+        event_dir = secure_filename(self.event_new.title, None)
         segments = [event_dir] if event_dir else []
+        if self.ALLOW_UNSCHEDULED and _get_start_dt(attachment.folder.object) is None:
```

Comment:
why check for allow_unscheduled again? I guess you only get here in case it's true if a contribution is not scheduled?